src/Lean/Elab/Quotation.lean

```diff
@@ -296,14 +320,19 @@ private def adaptRhs (fn : Term → TermElabM Term) : Alt → TermElabM Alt
 
 private partial def getHeadInfo (alt : Alt) : TermElabM HeadInfo :=
   let pat := alt.fst.head!
-  let unconditionally (rhsFn) := pure {
+  let unconditionally rhsFn := pure {
     check := unconditional,
     doMatch := fun yes _ => yes [],
     onMatch := fun taken => covered (adaptRhs rhsFn ∘ noOpMatchAdaptPats taken) (match taken with | unconditional => true | _ => false)
   }
   -- quotation pattern
-  if isQuot pat then
+  if isQuot pat then do
     let quoted := getQuotContent pat
+    discard <| getQuotKind pat
```

Comment:
This line needs a comment. I think it would also be better to move it up into `match_syntax.expand `.